### ISSUE TITLE
Semgrep fails with an SSL error

### ISSUE THREAD

**Issue by drorb** [semgrep/semgrep#1831]
Trying to run semgrep with a ruleset from the registry fails with an SSL related error.

When running the following command

`semgrep --config "https://semgrep.dev/p/r2c-security-audit"`

I get the following error

```
A new version of Semgrep is available. Please see https://github.com/returntocorp/semgrep#upgrading for more information.
using config from https://semgrep.dev/p/r2c-security-audit. Visit https://semgrep.dev/registry to see all public rules.
downloading config...
Failed to download config from https://semgrep.dev/p/r2c-security-audit: HTTPSConnectionPool(host='semgrep.dev', port=443): Max retries exceeded with url: /p/r2c-security-audit (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1124)')))
no valid configuration file found (1 configs were invalid)
```

Using semgrep version 0.27.0 installed using brew on OSX Catalina 10.15.7

**Comment by aryx** [comment]
@DrewDennison any idea?

**Comment by ievans** [comment]
@drorb thanks for making an issue! I can't reproduce this. It seems like the issue might be with your local certificate chain. Can you report whether 

`curl -L -H 'User-Agent: Semgrep/0.27' https://semgrep.dev/p/r2c-security-audit`

works for you?

**Comment by drorb** [comment]
I can confirm that `curl -L -H 'User-Agent: Semgrep/0.27' https://semgrep.dev/p/r2c-security-audit` works for me

**Comment by ievans** [comment]
@drorb thanks for trying that.

In the meantime as a workaround -- you've probably already done this, but just to be explicit:
```
curl -L -H 'User-Agent: Semgrep/0.27' https://semgrep.dev/p/r2c-security-audit > r2c-security-audit.yaml
semgrep --config r2c-security-audit.yaml
```

I'm at a bit of a loss as to where the issue could be. What is your specific Python version? cc @nbrahms @brendongo for ideas on the root cause.

**Comment by nbrahms** [comment]
Hi @drorb ...

Can you run:
```shell
/path/to/brew/python3 -m pip show certifi
```

where `/path/to/brew/python3` is probably something analogous to `/usr/local/Cellar/python@3.8/3.8.5/bin/python3` for you.

Please make sure the version is at least `2020.6.20`:
```
Name: certifi
Version: 2020.6.20
Summary: Python package for providing Mozilla's CA Bundle.
Home-page: https://certifiio.readthedocs.io/en/latest/
Author: Kenneth Reitz
Author-email: me@kennethreitz.com
License: MPL-2.0
Location: /usr/local/lib/python3.8/site-packages
Requires:
Required-by: requests, pipenv
```

If it's not, you should be able to run
```shell
/path/to/brew/python3 -m pip install --upgrade certifi
```

Hope this helps 🤞 .

**Comment by nbrahms** [comment]
@ievans It's my understanding that the python `requests` library uses the certificate authority from the `certifi` package, not any locally installed `openssl`, etc. CA.

**Comment by nbrahms** [comment]
@ievans Also looking through the dependencies of `semgrep`, we appear to be using `requests 2.22.0`, which itself only requires `certifi>=2017.4.17`. We may want to advance the `requests` or `certifi` dependency manually within `semgrep`.

**Comment by nbrahms** [comment]
Although it's worth noting the signer cert is from 2015:

```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            06:7f:94:57:85:87:e8:ac:77:de:b2:53:32:5b:bc:99:8b:56:0d
    Signature Algorithm: sha256WithRSAEncryption
        Issuer: C=US, O=Amazon, CN=Amazon Root CA 1
        Validity
            Not Before: Oct 22 00:00:00 2015 GMT
            Not After : Oct 19 00:00:00 2025 GMT
        Subject: C=US, O=Amazon, OU=Server CA 1B, CN=Amazon
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
                Modulus:
                    00:c2:4e:16:67:dd:ce:bc:6a:c8:37:5a:ec:3a:30:
                    b0:1d:e6:d1:12:e8:12:28:48:cc:e8:29:c1:b9:6e:
                    53:d5:a3:eb:03:39:1a:cc:77:87:f6:01:b9:d9:70:
                    cc:cf:6b:8d:e3:e3:03:71:86:99:6d:cb:a6:94:2a:
                    4e:13:d6:a7:bd:04:ec:0a:16:3c:0a:eb:39:b1:c4:
                    b5:58:a3:b6:c7:56:25:ec:3e:52:7a:a8:e3:29:16:
                    07:b9:6e:50:cf:fb:5f:31:f8:1d:ba:03:4a:62:89:
                    03:ae:3e:47:f2:0f:27:91:e3:14:20:85:f8:fa:e9:
                    8a:35:f5:5f:9e:99:4d:e7:6b:37:ef:a4:50:3e:44:
                    ec:fa:5a:85:66:07:9c:7e:17:6a:55:f3:17:8a:35:
                    1e:ee:e9:ac:c3:75:4e:58:55:7d:53:6b:0a:6b:9b:
                    14:42:d7:e5:ac:01:89:b3:ea:a3:fe:cf:c0:2b:0c:
                    84:c2:d8:53:15:cb:67:f0:d0:88:ca:3a:d1:17:73:
                    f5:5f:9a:d4:c5:72:1e:7e:01:f1:98:30:63:2a:aa:
                    f2:7a:2d:c5:e2:02:1a:86:e5:32:3e:0e:bd:11:b4:
                    cf:3c:93:ef:17:50:10:9e:43:c2:06:2a:e0:0d:68:
                    be:d3:88:8b:4a:65:8c:4a:d4:c3:2e:4c:9b:55:f4:
                    86:e5
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Basic Constraints: critical
                CA:TRUE, pathlen:0
            X509v3 Key Usage: critical
                Digital Signature, Certificate Sign, CRL Sign
            X509v3 Subject Key Identifier:
                59:A4:66:06:52:A0:7B:95:92:3C:A3:94:07:27:96:74:5B:F9:3D:D0
            X509v3 Authority Key Identifier:
                keyid:84:18:CC:85:34:EC:BC:0C:94:94:2E:08:59:9C:C7:B2:10:4E:0A:08

            Authority Information Access:
                OCSP - URI:http://ocsp.rootca1.amazontrust.com
                CA Issuers - URI:http://crt.rootca1.amazontrust.com/rootca1.cer

            X509v3 CRL Distribution Points:

                Full Name:
                  URI:http://crl.rootca1.amazontrust.com/rootca1.crl

            X509v3 Certificate Policies:
                Policy: 2.23.140.1.2.1```

**Comment by drorb** [comment]
The result of running `python3 -m pip show certifi` is

```
Name: certifi
Version: 2020.6.20
Summary: Python package for providing Mozilla's CA Bundle.
Home-page: https://certifiio.readthedocs.io/en/latest/
Author: Kenneth Reitz
Author-email: me@kennethreitz.com
License: MPL-2.0
Location: /usr/local/lib/python3.8/site-packages
Requires:
Required-by:
```

**Comment by nbrahms** [comment]
@drorb ... well that is super confusing. Can you confirm that `semgrep` is running using the python binary at your `/usr/local/lib/python3.8/` install?

**Comment by drorb** [comment]
@nbrahms so it seems that the brew installation of semgrep comes with its own python located in `/usr/local/Cellar/semgrep/0.27.0/libexec/bin/python3.8`
Looking at the semgrep script it seems to be using that version - `#!/usr/local/Cellar/semgrep/0.27.0/libexec/bin/python3.8`

The results of running `/usr/local/Cellar/semgrep/0.27.0/libexec/bin/python3 -m pip show certifi` are:

```
Name: certifi
Version: 2020.6.20
Summary: Python package for providing Mozilla's CA Bundle.
Home-page: https://certifiio.readthedocs.io/en/latest/
Author: Kenneth Reitz
Author-email: me@kennethreitz.com
License: MPL-2.0
Location: /usr/local/Cellar/semgrep/0.27.0/libexec/lib/python3.8/site-packages
Requires:
Required-by: requests
```

**Comment by nbrahms** [comment]
Thanks @drorb . Alas, I don't think we can reproduce on our end. We'll keep an eye out, and look at pinning cert versions on our end.

**Comment by nbrahms** [comment]
For posterity: we've been unable to reproduce this issue, so we're closing it out until we can 😞

**Comment by sigvissec** [comment]
I ran into this same issue while on MacOS on a corporate network and found a solution. As stated previously, semgrep ships with Python and relies upon certifi for certificate validation. 

One method to resolve this is to export the certificate used for mitm-ing the traffic and then to add it to the cacert.pem file within the certifi directory. To be clear, I already have that certificate added to by keychain and run into no issues with other tools, CLI based or otherwise, but for some reason with this distribution the keychain is not leveraged when performing certificate validation.

For me, with a Homebrew install of semgrep, I added the exported cert to /usr/local/Cellar/semgrep/\<version\>/libexec/lib/python\<version\>/site-packages/certifi/cacert.pem.
